src/main/java/seedu/contax/logic/parser/RangeCommandParser.java

```diff
@@ -31,13 +30,20 @@ public RangeCommand parse(String args) throws ParseException {
         Index toIndex;
 
         if (argMultimap.getValue(PREFIX_RANGE_FROM).isPresent() && argMultimap.getValue(PREFIX_RANGE_TO).isPresent()) {
-            fromIndex = Index.fromOneBased(Integer.parseInt(argMultimap.getValue(PREFIX_RANGE_FROM).get()));
-            toIndex = Index.fromOneBased(Integer.parseInt(argMultimap.getValue(PREFIX_RANGE_TO).get()));
+            try {
+                fromIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_RANGE_FROM).get());
+                toIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_RANGE_TO).get());
+            } catch (ParseException pe) {
+                throw new ParseException(String.format(MESSAGE_INVALID_RANGE_INDEX,
+                        RangeCommand.MESSAGE_USAGE), pe);
+            }
             if (fromIndex.getZeroBased() > toIndex.getZeroBased()) {
-                throw new ParseException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+                throw new ParseException(String.format(MESSAGE_INVALID_RANGE_INDEX,
+                        RangeCommand.MESSAGE_USAGE));
```

Comment:
I think you should explicitly state the error (from > to), rather than a generic message.

Reply:
thx changed